### PR TITLE
Added hackbench test to avocado

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Signed-off-by: praveen@linux.vnet.ibm.com
added hackbench Test case

### CODE REVIEW

File: perf/hackbench.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            cc = 'cc'
+        process.system('%s  hackbench.c -o hackbench -lpthread' % cc)
+
+    def test(self):
+
+        num_groups = self.params.get('num_groups', default=90)
+        hackbench_bin = os.path.join(self.srcdir, 'hackbench')
+        cmd = '%s %s' % (hackbench_bin, num_groups)
+        raw_output = process.system_output(cmd, shell=True)
+        self.results = raw_output
```
Comment (apahim) [inline perf/hackbench.py:68]: why not `self.results = process.system_output(cmd, shell=True)`?

File: perf/hackbench.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        perf_json = {}
+        lines = self.results.split('\n')
+        for line in lines:
```
Comment (apahim) [inline perf/hackbench.py:71]: why not `for line in self.results.split('\n')`?

**Comment by apahim** [comment]
Is there a condition that could be used to consider the test as `FAIL`? I mean, we are running `hackbench` and collecting the time... what is a bad result? Maybe a threshold in time?

**Comment by PraveenPenguin** [comment]
@apahim  threshold  time of hackbanch differs distro to distro 

as example below ubuntu behaviour as below 

```
   user@host: ~ $ hackbench
   Running in process mode with 10 groups using 40 file  descriptors  each
   (== 400 tasks)
   Each sender will pass 100 messages of 100 bytes
   Time: 0.890
```

so not sure which can be common for all distribution  as well as upto groups so it up to user to after run and analyze the result .It also 

can please help me on this .

Thanks in advance

File: perf/hackbench.py (hunk 2 above)
Comment (PraveenPenguin) [inline perf/hackbench.py:71]: @apahim  thanks will interoperate all the suggestion and raise a new PR.

Thanks

**Comment by PraveenPenguin** [comment]
@apahim  can please help me  threshold time , if you agree what is proposed earlier that user need identify  the failure , will request new PR with changes

Thanks

**Comment by apahim** [comment]
@PraveenPenguin the threshold could then be configurable, defaults to no threshold.

**Comment by PraveenPenguin** [comment]
@apahim  thanks for input , will update code with suggested change and open in new PR

Thanks

**Comment by PraveenPenguin** [comment]
@apahim added new PR #116   with change 

Thanks

**Pull request closed without merging**